Dmf/Modules.Library/Dmf_String.c

```diff
@@ -114,49 +114,118 @@ Return Value:
 
 --*/
 {
+    LONG returnValue;
+    size_t comparisonLength;
+
     UNREFERENCED_PARAMETER(DmfModule);
 
-    return strncmp(StringInList,
-                   LookFor,
-                   strlen(StringInList));
+    // Always check full length of given string.
+    //
+    comparisonLength = strlen(LookFor);
+
+    if (0 == comparisonLength)
+    {
+        // Special case because strncmp always returns 0 in this case. It happens if
+        // either string is "".
+        //
+        if (*StringInList)
+        {
+            // Given string is smaller.
+            //
+            returnValue = -1;
+        }
+        else
+        {
+            // They are both equal.
+            //
+            returnValue = 0;
+        }
+        goto Exit;
+    }
+
+    returnValue = strncmp(StringInList,
+                          LookFor,
+                          comparisonLength);
+
+Exit:
+
+    return returnValue;
 }
 
-LONG
-String_FindInListLeftLookForMatchCharCallback(
-    _In_ DMFMODULE DmfModule,
-    _In_ CHAR* StringInList,
-    _In_ CHAR* LookFor
+#if defined(DMF_USER_MODE)
+
+static
+WCHAR* 
+String_MultiStringToWideString(
+    _In_ CHAR* NarrowString
     )
 /*++
 
 Routine Description:
 
-    Perform LEFT case sensitive comparison between string in list and the given string.
-    Comparison is made between the full given string and left of the string in the list.
+    Allocate a buffer for a Wide string and copy into it a converted version of a given

```

Comment:
I would recommend documenting that it is the caller's responsibility to call free() on the string returned.

Reply:
Yes, I will do that on next update.